src/ngamsServer/ngamsServer/commands/status.py

```diff
@@ -515,19 +525,31 @@ def handleCmd(srvObj,
         genDiskStatus = 1
     elif (fileId):
         if (not fileVersion): fileVersion = -1
-        fileObj = ngamsFileInfo.ngamsFileInfo()
-        fileObj.read(srvObj.getHostId(), srvObj.getDb(), fileId, fileVersion)
-        diskObj = ngamsDiskInfo.ngamsDiskInfo()
         try:
-            diskObj.read(srvObj.getDb(), fileObj.getDiskId())
+            fileObj = ngamsFileInfo.ngamsFileInfo()
+            fileObj.read(srvObj.getHostId(), srvObj.getDb(), fileId, fileVersion)
+            diskObj = ngamsDiskInfo.ngamsDiskInfo()
+            try:
+                diskObj.read(srvObj.getDb(), fileObj.getDiskId())
+            except:
+                errMsg = "Illegal Disk ID found: {0} for file with ID: {1}".format(fileObj.getDiskId(), fileId)
+                raise Exception(errMsg)
+            diskObj.addFileObj(fileObj)
+            status.addDiskStatus(diskObj)
+            genDiskStatus = 1
+            genFileStatus = 1
         except:
-            errMsg = "Illegal Disk ID found: %s for file with ID: %s" %\
-                     (fileObj.getDiskId(), fileId)
-            raise Exception(errMsg)
-        diskObj.addFileObj(fileObj)
-        status.addDiskStatus(diskObj)
-        genDiskStatus = 1
-        genFileStatus = 1
+            # The file was not found in the database. Check if it is available
+            # on a remote partner site.
+            host, port, status_info, disk_info, file_info = \
+                ngamsFileUtils.lookup_partner_site_file_status(srvObj, fileId,
+                                                               fileVersion,
+                                                               reqPropsObj)
+            # Update status reply using the disk status and file status
+            # information retrieved from the partner site
+            status.addDiskStatus(disk_info)
+            genDiskStatus = 1
+            genFileStatus = 1
```

Comment:
These are set both in case of success of failure, so they could be set only once after the `try`/`except` block.

Reply:
I will fix this